pkg/build/api/types.go

```diff
@@ -76,8 +76,73 @@ type BuildSpec struct {
 	// scheduled in the system, that the build may be active on a node before the
 	// system actively tries to terminate the build; value must be positive integer
 	CompletionDeadlineSeconds *int64
+
+	// Info is a placeholder that can store multiple reasons why a build was
+	// triggered. It should cover for a future implementation of level triggered
+	// builds where you can multiple reasons why a build was triggered.
+	Info []*BuildTriggerInfo
```

Comment:
> type BuildConfigSpec {
  CommonSpec
  Triggers []BuildTriggerPolicy
}
This causes problems because this right now is
> // BuildConfigSpec describes when and how builds are created
type BuildConfigSpec struct {
	// Triggers determine how new Builds can be launched from a BuildConfig. If no triggers
	// are defined, a new build can only occur as a result of an explicit client build creation.
	Triggers []BuildTriggerPolicy 

	// BuildSpec is the desired build specification
	BuildSpec 
}